cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/CucumberExpressionGeneratorTest.java

```diff
@@ -234,43 +273,6 @@ public void exposes_transforms_in_generated_expression() {
         assertEquals(Double.class, generatedExpression.getParameterTypes().get(1).getType());
     }
 
-    @Test
-    public void matches_parameter_types_with_optional_capture_groups() {
-        ParameterType<String> optionalFlight = new ParameterType<>(
-                "optional-flight",
-                "(1st flight)?",
-                String.class,
-                new Transformer<String>() {
-                    @Override
-                    public String transform(String arg) {
-                        return arg;
-                    }
-                },
-                true,
-                false
-        );
-        ParameterType<String> optionalHotel = new ParameterType<>(
-                "optional-hotel",
-                "(1st hotel)?",
-                String.class,
-                new Transformer<String>() {
-                    @Override
-                    public String transform(String arg) {
-                        return arg;
-                    }
-                },
-                true,
-                false
-        );
-
-        parameterTypeRegistry.defineParameterType(optionalFlight);
-        parameterTypeRegistry.defineParameterType(optionalHotel);
-        // While you would expect this to be `I reach Stage{int}: {optional-flight}-{optional-hotel}` the `-1` causes
-        // {int} to match just before {optional-hotel}.
-        List<GeneratedExpression> generatedExpressions = generator.generateExpressions("I reach Stage4: 1st flight-1st hotel");
-        assertEquals("I reach Stage{int}: {optional-flight}{int}st hotel", generatedExpressions.get(0).getSource());
```

Comment:
I'm would not expect this to be deleted. This should now match `"I reach Stage{int}: {optional-flight}{optional-hotel}" because the `-` is punctuation.

Reply:
You're right @mpkorstanje - deleting it was perhaps a bit sloppy of me. The thing is, I couldn't make the test verify the behaviour it was initially written to test, namely that the optional group had a 0-length match. We could keep it, tweak it and make it pass as you suggested, but it would be testing behaviour we're already testing I think...

Reply:
ON the Ruby implementation, I've had to update this test so it still reflects the expected behaviour (matching the int before the optional capture group).
The sentence does not make that much sense now, hopefully the comment below still explain correctly why we don't match (optional-hotel)